MCR-2200 add filter to enable content negotation by extension

rsteph-de opened the pull request:
[Link to jira](https://mycore.atlassian.net/browse/MCR-2200).

rsteph-de committed:
MCR-2200 add filter to enable content negotation by extension

MyCoRe-Travis commented:
Something gone wrong on 7beeec0710e844bfde1b376460e515a54b1bfbea! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5520-pull_request-964

yagee-de commented on mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java:
```diff
@@ -0,0 +1,61 @@
+
+    private static final Map<String, MediaType> MEDIA_TYPE_MAPPINGS = Map.ofEntries(
+            new AbstractMap.SimpleEntry<String, MediaType>("json", MediaType.APPLICATION_JSON_TYPE),
+            new AbstractMap.SimpleEntry<String, MediaType>("xml", MediaType.APPLICATION_XML_TYPE));
```
should be `Map.entry(key, value)`

rsteph-de replied:
O.K.

yagee-de commented on mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java:
```diff
@@ -0,0 +1,61 @@
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        if(!requestContext.getUriInfo().getPath().contains("/contents/")) {
```
Better use a property with regex filter

rsteph-de replied:
done

yagee-de commented on mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestV2App.java:
```diff
@@ -50,8 +51,11 @@
 
     public MCRRestV2App() {
         super();
+        if (MCRConfiguration2.getBoolean("MCR.RestAPI.V2.EnableContentNegotiationViaFileExtension").orElse(false)) {
```
We should always enable it (can be excluded by regex: see above)

rsteph-de replied:
done

rsteph-de committed:
MCR-2200 improvements after review (check URI by regex)

yagee-de commented:
How does this Filter interfere with MCRNormalizeMCRObjectIDsFilter?

rsteph-de commented:
> 
> 
> How does this Filter interfere with MCRNormalizeMCRObjectIDsFilter?

Conceptually it should not. The extension *.xml / *.json should be removed first and after that the  MCRNormalizeMCRObjectIDsFilter should be applied. - I hope the code "works as designed".
Let's check the order of filters again.

rsteph-de committed:
MCR-2200 improve @Priority of Filters

MCRContentNegotationViaExtensionFilter and
MCRNormalizeMCRObjectsIDsFilter are executed in the required order.

adlerre commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java  2
- mycore-restapi/src/main/java/org/mycore/restapi/MCRNormalizeMCRObjectIDsFilter.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5511336&bid=18062296)
